### PR TITLE
statistics: change permission at the library level

### PR DESCRIPTION
**Pull request by jma**
* Adds category, indicator, frequence and library facets.
* Changes the stat config update, delete permission at the library level.
* Collects the librarian stats during the setup.
* Adds pickup location distribution for requests.
* Adds owning location distrubution for circulation actions.
* Removes period for ill request report stats.

Co-Authored-by: Johnny Mariéthoz <Johnny.Mariethoz@rero.ch>

### CODE REVIEW

File: rero_ils/modules/stats_cfg/api.py [hunk 1]
```diff
@@ -92,16 +95,28 @@ def reasons_not_to_delete(self):
...
         # It is not possible to delete configuration if there are reports.
-        links = self.get_links_to_me(self.pid)
+        links = self.get_links_to_me()
         if links:
             cannot_delete['links'] = links
         return cannot_delete
 
+    @property
+    def organisation_pid(self):
+        """Shortcut for acquisition order organisation pid."""
```
Comment (PascalRepond) [inline rero_ils/modules/stats_cfg/api.py:105]: acquisition order -> "stat config"

File: rero_ils/modules/stats_cfg/api.py [hunk 2]
```diff
@@ -92,16 +95,28 @@ def reasons_not_to_delete(self):
...
+        library = extracted_data_from_ref(self.get('library'), data='record')
+        return library.organisation_pid
+
+    @property
+    def library_pid(self):
+        """Shortcut for acquisition order library pid."""
```
Comment (PascalRepond) [inline rero_ils/modules/stats_cfg/api.py:111]: Same

**Review by PascalRepond — APPROVED** [review]

**Review by rerowep — COMMENTED** [review]
Do we need an alembic for production?

File: rero_ils/modules/stats/extensions.py [hunk 3]
```diff
@@ -41,8 +43,15 @@ def pre_dump(self, record, data, dumper=None):
         :param dumper: the dumper class used to dump the record.
         """
         # to filter the search list results
-        if org := record.get('config', {}).get('organisation'):
-            record['organisation'] = org
+        if lib := record.get('config', {}).get('library'):
+            if lib.get('pid'):
+                lib_pid = lib['pid']
+            else:
+                lib_pid = extracted_data_from_ref(lib.get('$ref'))
```
Comment (zannkukai) [inline rero_ils/modules/stats/extensions.py:50]: `lib_pid = lib.get('pid') or extracted_data_from_ref(lib.get('$ref'))`

OR 

`lib_pid = lib.get('pid', extracted_data_from_ref(lib.get('$ref'))`

**Review by zannkukai — APPROVED** [review]

**Review by PascalRepond — APPROVED** [review]

**Pull request merged**